### PR TITLE
Add support for release-drafter

### LINKED ISSUE #233: Add support for release-drafter
This issue is to track adding support for release-drafter. With release drafter the changelog of esch release is automatically generated reducing the burden of creating releases. This would also allow users to know what changed during each release.

@runabol I can work on this during the weekend, I already have the templates/configs.

### PR DESCRIPTION
**Pull request by gaby**
Fixes #233 

@runabol How do you want to handle `minor` and `major` tags? Release drafter will auto-generate the tag based on the labels of the PR's that have been merged.

```yaml
version-resolver:
  major:
    labels:
      - 'major'
  minor:
    labels:
      - 'minor'
      - '❗ Breaking Change'
  patch:
    labels:
      - '✏️ Feature'
      - '📒 Documentation'
      - '🐛 Bug'
      - '🤖 Dependencies'
      - '🧹 Updates'
      - '🔒 Security'
```

In my initial draft, I have left both `major`  and `minor` as place holders. All the other labels will increment the `patch` release. Do you want to move some of those or add new ones to manage the `minor` releases?

### CODE REVIEW

**Comment by runabol** [comment]
@gaby this is great. So will this automatically kick in whenever I merge to `main` and use the title of the commit to update the version? What about the `version.go` file which I use at build time?

**Comment by gaby** [comment]
@runabol This does several things:

- When a PR is created, it will auto-label the PR based on keywords specified at the bottom of the `release-drafter.yml` file
- When a PR is merged to `main`, it will trigger the github-action that will auto-generate the release notes, you can still manually edit the release and then publish.
- It will not update the `versions.go` file, that's still up to you.

TLDR: For each merged into `main` it will auto-generate a draft release based on the labels of each PR merged since the last release.

**Comment by runabol** [comment]
Let's try it!

**Comment by gaby** [comment]
@runabol It will start working once another PR is created, since mine didnt have any labels assigned

**Comment by runabol** [comment]
Yea, I see now how it works:

Screenshot 2023-10-30 at 12 35 15 AM

Really cool!

**Comment by gaby** [comment]
@runabol You are suppose to click edit on the draft and publisb it. Else it wont have notes 😁

**Comment by runabol** [comment]
Yea, I think I got it now :-)

**Comment by runabol** [comment]
@gaby this feature is a game changer! thanks again!

**Comment by gaby** [comment]
@runabol Glad you find it useful! 💪 it helps a lot to reduce the overhead of doing open source work
